### PR TITLE
Bugfix: response transformer

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change
Handling for failure cases
Wrapping Status code
Code clean up

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by sanpj2292 — CHANGES_REQUESTED** [review]

File: v0/destinations/braze/networkResponseHandler.js [hunk 1]
```diff
@@ -1,26 +1,48 @@
...
-      .setStatus(400)
-      .setMessage("Braze Request Failed")
-      .setDestinationResponse({
-        response: respBody,
-        status: destResponse.status
+      .setStatus(status)
+      .setMessage(
+        `[Braze Response Transfom] Request failed for ${DESTINATION} with status: ${status}`
+      )
+      .setDestinationResponse(destinationResponse)
```
[the remark this answers is not in the document]
Comment (utsabc) [reply v0/destinations/braze/networkResponseHandler.js:28]: But for exceptions we do not throw/ thrown by the system we will not have the destination response hence was setting that in `versionedRouter.js`

File: v0/util/index.js [hunk 2]
```diff
@@ -1087,6 +1053,17 @@ function populateErrStat(error, destination, isStageTransform = true) {
   }
   return error;
 }
+/**
+ * Returns true for http status code in range of 200 to 300
+ * @param {*} status
+ * @returns
+ */
+function isHttpStatusSuccess(status) {
+  if (status >= 200 && status < 300) {
```
[the remark this answers is not in the document]
Comment (utsabc) [reply v0/util/index.js:1062]: Will address this

File: v0/destinations/braze/networkResponseHandler.js [hunk 3]
```diff
@@ -1,26 +1,48 @@
...
 const { TRANSFORMER_METRIC } = require("../../util/constant");
 const { DESTINATION } = require("./config");
 const ErrorBuilder = require("../../util/error");
 
-const responseTransform = destResponse => {
-  let respBody;
+// eslint-disable-next-line no-unused-vars
+const responseTransform = (destResponse, _dest) => {
+  let response;
   try {
```
[the remark this answers is not in the document]
Comment (utsabc) [reply v0/destinations/braze/networkResponseHandler.js:10]: This was missed fixed it

File: __tests__/data/braze_response_output.json [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        "errors": [
+          {
+            "code": "604"
+          }
+        ],
+        "message": "failed"
+      },
+      "status": 200
+    },
+    "message": "[Braze Response Transfom] Request failed for braze with status: 200",
```
Comment (sanpj2292) [inline __tests__/data/braze_response_output.json:15]: `[Braze Response Transfom] Request failed for braze with status: 200`
This is confusing IMO. May be showing it as
`[Braze Response Transfom] Request failed for braze with status: 604` would be better I guess

File: package.json [hunk 5]
```diff
@@ -22,8 +22,9 @@
     "start": "node destTransformer.js",
     "testDestinations": "npx jest --runInBand --detectOpenHandles --coverage --notify",
     "testVersionedRouter": "npx jest versionedRouter --type=all --detectOpenHandles --notify",
-    "testVersioned": "npx jest versionedRouter --destName=braze --type=batch",
-    "verify": "eslint . || exit 1; jest --runInBand --detectOpenHandles --coverage --notify || exit 1"
+    "testVersioned": "npx jest versionedRouter --destName=heap --type=all --notify",
+    "verify": "eslint . || exit 1; jest --runInBand --detectOpenHandles --coverage --notify || exit 1",
+    "coverage": "npx jest --runInBand --detectOpenHandles --coverage --notify && npx jest versionedRouter --destName=heap --type=all --detectOpenHandles  --notify"
```
Comment (sanpj2292) [inline package.json:27]: > npx jest versionedRouter --destName=heap --type=all --detectOpenHandles  --notify

Why do we need this for heap specifically ?

File: v0/destinations/braze/networkResponseHandler.js [hunk 6]
```diff
@@ -1,47 +1,52 @@
...
+        `[Braze Response Transfom] Request failed for ${DESTINATION} with status: ${status}`
+      )
+      .setDestinationResponse(destinationResponse)
       .isTransformResponseFailure(true)
       .setStatTags({
         destination: DESTINATION,
         scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
         stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.RESPONSE_TRANSFORM,
-        meta: getDynamicMeta(destResponse.status || 400)
+        meta: getDynamicMeta(status)
```
Comment (sanpj2292) [inline v0/destinations/braze/networkResponseHandler.js:42]: Is it `400` here as well ?

File: package.json (hunk 5 above)
Comment (utsabc) [reply package.json:27]: We have written the versionedRouter test for heap as the default destination. This was original implementation of versionedRouter test

File: v0/destinations/braze/networkResponseHandler.js (hunk 6 above)
Comment (utsabc) [reply v0/destinations/braze/networkResponseHandler.js:42]: As we always expect status from the destination, we are not setting 400 as a fallback here as we already have it at server level

File: __tests__/data/braze_response_output.json (hunk 4 above)
Comment (utsabc) [reply __tests__/data/braze_response_output.json:14]: This is addressed

**Review by sanpj2292 — APPROVED** [review]
LGTM

**Pull request merged**